src/reynir_correct/pattern.py

```diff
@@ -1395,43 +1419,44 @@ def create_patterns(cls) -> None:
                     None,
                 )
             )
-            # Catch "Hann leiðir líkur af því."
+
+            # Catch "Hann leiðir ekki líkur af því."
             p.append(
                 (
                     "leiða",  # Trigger lemma for this pattern
-                    "VP > { VP > { VP > { 'leiða' } NP-OBJ > { ('líkur' | 'rök') } } PP > { P > { 'af' } } }",
+                    "VP > { VP > { 'leiða' } NP-PRD > { ('líkur' | 'rök' | 'rak') } PP > { P > { 'af' } } } ",
                     cls.wrong_preposition_leida_af,
                     None,
                 )
             )
-            # Catch "Hann leiðir ekki líkur af því."
+            # Catch "Hann leiðir líkur af því.", "Hann leiðir rök af því.", "Hann hefur aldrei leitt líkur af því."
             p.append(
                 (
                     "leiða",  # Trigger lemma for this pattern
-                    "VP > { VP > { 'leiða' } NP-PRD > { ('líkur' | 'rök') } PP > { P > { 'af' } } } ",
+                    "VP > { VP >> { VP > { 'leiða' } NP > { ('líkur' | 'rök' | 'rak') } } PP > { 'af' } } ",
                     cls.wrong_preposition_leida_af,
                     None,
                 )
             )
-            # Catch "Hann hefur aldrei leitt líkur af því."
+            # Catch "Tíminn markar upphaf af því."
             p.append(
                 (
-                    "leiða",  # Trigger lemma for this pattern
-                    "VP > { VP >> { VP > { 'leiða' } NP > { ('líkur' | 'rök') } } PP > { 'af' } } ",
-                    cls.wrong_preposition_leida_af,
+                    "upphaf",  # Trigger lemma for this pattern
+                    "VP > { VP > { 'marka' } NP-OBJ > { 'upphaf' PP > { 'af' } } }",
+                    cls.wrong_preposition_marka_af,
                     None,
                 )
             )
-            # Catch "Tíminn markar (ekki) upphaf af því."
+            # Catch "Það markar ekki upphaf af því."
             p.append(
                 (
-                    "upphaf",  # Trigger lemma for this pattern
-                    "VP > { VP > { 'marka' } NP-OBJ > { 'upphaf' PP > { 'af' } } }",
+                    "upphafinn",  # Trigger lemma for this pattern
```

Comment:
We need to be careful here. "Upphafinn" is probably a wrong parse (wrong terminal and tag), the correct trigger lemma would be "upphaf" (hk noun). We should at least include both words as trigger lemmas.

Reply:
Done!